src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -295,7 +304,13 @@ public CompilationTracker FreezePartialStateWithTree(SolutionState solution, Doc
                 }
 
                 inProgressProject = inProgressProject.AddProjectReferences(newProjectReferences);
-                inProgressCompilation = UpdateCompilationWithNewReferencesAndRecordAssemblySymbols(inProgressCompilation, metadataReferences, metadataReferenceToProjectId);
+
+                if (!Enumerable.SequenceEqual(inProgressCompilation.ExternalReferences, metadataReferences))
```

Comment:
what's an ExtenralReference?

Reply:
It's the list of references that excludes references that get added via `#r`. Those are computed automatically and added into the Compilation's direct "References" collection, but in this case we're explicitly only mutating the ones that came elsewhere.

(This code just moved, but it's a great question nonetheless.)